frontend/src/main/web/lib/utils/GlossaryHelper.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash';
+import StringUtils from './StringUtils'
+
+var GlossaryHelper = {
+  /**
+   * Generate org.zanata.rest.dto.GlossaryTerm object
+   * @param data
+   */
+  generateGlossaryTermDTO: function (data) {
+    if(_.isUndefined(data)) {
+      return null;
+    }
+    var content = StringUtils.trim(data.content),
+      locale = data.locale,
+      comments = StringUtils.trim(data.comment);
+
+    if(StringUtils.isEmptyOrNull(content) &&
+        StringUtils.isEmptyOrNull(locale)) {
+      return null;
+    } else {
+      var term = {};
+      term['content'] = content;
+      term['locale'] = locale;
+      term['comments'] = comments;
+      return term;
+    }
+  },
+
+  /**
+   * Generate org.zanata.rest.dto.Glossary object
+   * @param data
+   */
+  generateGlossaryDTO: function (data) {
+    var glossary = {}, entry = {};
+    glossary['glossaryEntries'] = [];
+
+    entry['resId'] = data.resId;
+    entry['pos'] = StringUtils.trim(data.pos);
+    entry['description'] = StringUtils.trim(data.description);
+    entry['srcLang'] = data.srcTerm.locale;
+    entry['sourceReference'] = data.srcTerm.reference;
+    entry['glossaryTerms'] = [];
+
+    var srcTerm = this.generateGlossaryTermDTO(data.srcTerm);
+    if(!_.isNull(srcTerm)) {
```

Comment:
This will execute if the value of srcTerm is `undefined`. Seems a bit risky to use `null` specifically - at least document it as the specific value that is used to indicate that nothing was generated.